### PR TITLE
Note about storage compatibility mode

### PR DESCRIPTION
**Pull request by fulldecent**
For #12547

### CODE REVIEW

File: docs/internals/layout_in_storage.rst [hunk 1]
```diff
@@ -56,7 +56,8 @@ as individual values.
      of Solidity due to the fact that storage pointers can be passed to libraries. This means that
      any change to the rules outlined in this section is considered a breaking change
      of the language and due to its critical nature should be considered very carefully before
-     being executed.
+     being executed. In the event of such a breaking change, we would want to release a
+     compatability mode which supports these the existing rules here.
```
Comment (cameel) [inline docs/internals/layout_in_storage.rst]:
```suggestion
     compatibility mode which supports the existing rules here.
```

**Review by cameel — CHANGES_REQUESTED** [review]

File: docs/internals/layout_in_storage.rst (hunk 1 above)
Comment (cameel) [reply docs/internals/layout_in_storage.rst]: I would also reword this a bit:
```suggestion
     compatibility mode in which the compiler would generate bytecode supporting the old layout.
```

**Review by cameel — DISMISSED** [review]

**Comment by chriseth** [comment]
Sounds good! Can you squash the two commits, please?

**Comment by fulldecent** [comment]
@chriseth squashed 🍞🥩🧀🥬➡️🍔

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Thanks!

**Pull request merged**